test/replaytests/workflows.go

```diff
@@ -698,3 +701,28 @@ func ResetWorkflowWithChild(ctx workflow.Context) (string, error) {
 	logger.Info("Child execution completed with result: " + result)
 	return result, nil
 }
+
+func WaitForCancelWorkflow(ctx workflow.Context, _ nexus.NoValue) (nexus.NoValue, error) {
+	return nil, workflow.Await(ctx, func() bool { return false })
+}
+
+var WaitOnSignalOp = temporalnexus.NewWorkflowRunOperation(
+	"wait-on-signal-op",
+	WaitForCancelWorkflow,
+	func(ctx context.Context, _ nexus.NoValue, soo nexus.StartOperationOptions) (client.StartWorkflowOptions, error) {
+		return client.StartWorkflowOptions{}, nil
+	},
+)
+
+func CancelNexusOperationWorkflow(ctx workflow.Context) error {
```

Comment:
We need to cover all possible previous state transitions we could have taken here basically:
* cancel before the operation start command is sent to the server
* cancel after the command is sent but before nexus operation is started
* cancel after the operation is started but before finished
* cancel after finished

We also need to make sure the workflow generates different commands on these different code paths so the SDK can detect the non determinism